neo/rawio/spikeglxrawio.py

```diff
@@ -0,0 +1,315 @@
+"""
+Class for reading data from a SpikeGLX system  (NI-DAQ for neuropixel probe)
+See https://billkarsh.github.io/SpikeGLX/
+
+
+Here an adaptation of the spikeglx tools into the neo rawio API.
+
+Note that each pair of ".bin"/."meta" represent a group of analog signal
+that share the same sampling rate.
+
+Contrary to other implementations this read the entire folder and subfolder so:
+  * It deal with severals segment taken from the naming "_gt0", "_gt1", "_gt2", ...
+  * It deal with all signal "imec0", "imec1" for neuropixel probes and also
+     external signal like"nidq" This is the "device"
+  * For imec device both "ap" and "lf" are extracted so one device have several "stream"
+
+Note:
+  * there are sevral version depending the NP probe generatio 1.0 2.0 3.0
+     I am not sure that the meta file is almost the same for all.
+     Maybe there some differences, this have to be check.
+     This IO have been tested basically wit neuropixel (NP) 2.0 one shank
+
+
+# TODO:
+  * contact SpkeGLX developer to see how to deal with absolut t_start when several segment
+  * contact SpkeGLX developer to understand the last channel SY0 function
+  * better handling of annotations at object level by sub group of device (need rawio change)
```

Comment:
rawio have a bad choice about annotation for AnalogSignal because it is done only at channel level (which is good for array_annotation) I need to add another dict for this. And also I want to make more explicit channel_indexes.
They are now global indexes and I would to introduce local (to stream) indexes in the APi to make dev more simple.